### ISSUE TITLE
MNIST_CNN example uses double softmax?

### ISSUE THREAD

**Issue by TimZaman** [fchollet/keras#6636]
Snippet from the Keras MNIST (LeNet) example ([source](https://github.com/fchollet/keras/blob/master/examples/mnist_cnn.py#L57)) :
```python
(...)
model.add(Dense(num_classes, activation='softmax'))
model.compile(loss=keras.losses.categorical_crossentropy,
(...)
```
Implementation of `categorical_crossentropy` uses `tf.nn.softmax_cross_entropy_with_logits()` ([source](https://github.com/fchollet/keras/blob/master/keras/backend/tensorflow_backend.py#L2750)):

But the Tensorflow doc on `tf.nn.softmax_cross_entropy_with_logits()` ([source](https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits)) states:
> **WARNING**: This op expects unscaled logits, since it performs a softmax on logits internally for efficiency. Do not call this op with the output of softmax, as it will produce incorrect results.

Am I missing something or are we violating the above warning due to `Dense()` calling the softmax activation?

**Comment by t-ae** [comment]
`keras.losses.categorical_crossentropy` can take result of a softmax.
https://github.com/fchollet/keras/blob/master/keras/backend/tensorflow_backend.py#L2731-L2732

**Comment by TimZaman** [comment]
Ah I see! I think combining them should be prefered (from_logits=True) as it should improve numerical stability.
